src/GravityForms/Controllers/ZaakController.php

```diff
@@ -31,98 +32,81 @@
  */
 class ZaakController extends AbstractZaakFormController
 {
-	public function handle(array $validation_result ): array
+	private ZaakUploadsController $uploads_controller;
+	private ZaakUploadPDFController $upload_pdf_controller;
+
+	public function __construct()
 	{
-		$this->set_class_properties( $validation_result['form'] );
+		parent::__construct();
+		$this->uploads_controller    = new ZaakUploadsController();
+		$this->upload_pdf_controller = new ZaakUploadPDFController();
+	}
 
-		if ( ! $this->form_is_zgw()) {
-			return $validation_result;
+	/**
+	 * Init Zaak creation and handle accordingly.
+	 */
+	public function handle( array $entry, array $form ): void
+	{
+		// Only create transaction for ZGW enabled forms.
+		if ( ! FormUtils::form_is_zgw( $form ) ) {
+			return;
 		}
 
+		// Make entry/form available to other methods that expect them.
+		$this->entry = $entry;
+		$this->form  = $form;
+
+		// Get the supplier name and key from form settings.
+		$supplier_name = FormUtils::supplier_form_setting( $form );
+		$supplier_key  = FormUtils::supplier_form_setting( $form, true );
+
 		try {
-			if ( ! count( $this->entry )) {
-				throw new Exception( $this->failed_messages['zaak'], 500 );
-			}
-
-			$this->handle_zaak_creation();
-		} catch (Exception $e) {
-			return $this->fail_form_validation(
-				validation_result: $validation_result,
-				failed_message: $e->getMessage()
-			);
-		}
+			// Create the Zaak.
+			$zaak = $this->handle_zaak_creation( $supplier_name, $supplier_key );
 
-		return $validation_result;
-	}
+			// Call the upload handlers to attach files to the Zaak.
+			$this->uploads_controller->set_form_data( $entry, $form );
+			$this->uploads_controller->handle( $zaak, $supplier_name, $supplier_key );
 
-	protected function set_failed_messages_property(): array
-	{
-		return array(
-			'zaak' => __(
-				'Er is een fout opgetreden bij het aanmaken van uw zaak. Probeer het later opnieuw.',
-				'owc-gravityforms-zgw'
-			),
-		);
+			// Call the PDF upload handler to attach documents to the Zaak.
+			$this->upload_pdf_controller->set_form_data( $entry, $form );
+			$this->upload_pdf_controller->handle( $zaak, $supplier_name, $supplier_key );
+
+			$this->mark_transaction_success( $zaak );
+		} catch ( \Throwable $e ) {
```

Comment:
Bovenaan `use Throwable;` en dan hier zonder de backslash. Maakt geen reet uit verder maar het is duidelijker welke classes er gebruikt worden en het is lijn met Exception.